src/stores/nodeStore.ts

```diff
@@ -0,0 +1,48 @@
+import { computed, observable, action } from 'mobx';
+import { INode, IRoutePath } from '../models';
+import NodeHelper from '../util/nodeHelper';
+
+export class NodeStore {
+    @observable private _nodes: INode[];
+
+    constructor() {
+        this._nodes = [];
+    }
+
+    @computed get nodes(): INode[] {
+        return this._nodes;
+    }
+
+    set nodes(value: INode[]) {
+        this._nodes = value;
+    }
+
+    public getNodes(nodeIds: number[]) {
```

Comment:
Why not just do `this.props.nodeStore!.nodes.filter(.....` in whatever is calling this? Or at least change the name to be more descriptive

Reply:
Probably not used